### ISSUE TITLE
有没有大佬能不能发一下跑成功的requirements.txt文件

### ISSUE THREAD

**Issue by Fg-droid-1014** [OpenBMB/MiniCPM-V#284]
### 是否已有关于该错误的issue或讨论？ | Is there an existing issue / discussion for this?

- [X] 我已经搜索过已有的issues和讨论 | I have searched the existing issues / discussions

### 该问题是否在FAQ中有解答？ | Is there an existing answer for this in FAQ?

- [X] 我已经搜索过FAQ | I have searched FAQ

### 运行环境 | Environment

```Markdown
- OS:
- Python:
- Transformers:
- PyTorch:
- CUDA (`python -c 'import torch; print(torch.version.cuda)'`):
```

**Comment by 1SingleFeng** [comment]
这是我的版本信息，仅供参考，我跑通了LORA微调的流程
Package                       Version
----------------------------- ------------
absl-py                       2.1.0
accelerate                    0.30.1
addict                        2.4.0
aiofiles                      23.2.1
altair                        5.2.0
annotated-types               0.6.0
anyio                         3.7.0
argon2-cffi                   21.3.0
argon2-cffi-bindings          21.2.0
astor                         0.8.1
asttokens                     2.2.1
async-lru                     2.0.2
attrs                         23.1.0
Babel                         2.12.1
backcall                      0.2.0
backports.functools-lru-cache 1.6.4
beautifulsoup4                4.12.2
bleach                        6.0.0
blis                          0.7.11
boltons                       23.0.0
brotlipy                      0.7.0
cachetools                    5.3.2
catalogue                     2.0.10
certifi                       2023.5.7
cffi                          1.15.1
charset-normalizer            2.0.4
click                         8.1.7
cloudpathlib                  0.16.0
cmake                         3.26.3
colorama                      0.4.6
conda                         23.3.1
conda-content-trust           0.1.3
conda-package-handling        2.0.2
conda_package_streaming       0.7.0
confection                    0.1.4
contourpy                     1.2.0
cryptography                  39.0.1
cycler                        0.12.1
cymem                         2.0.8
ddparser                      1.0.8
debugpy                       1.5.1
decorator                     5.1.1
deepspeed                     0.14.0
defusedxml                    0.7.1
dnspython                     2.3.0
editdistance                  0.6.2
einops                        0.7.0
entrypoints                   0.4
et-xmlfile                    1.1.0
etcd-config                   1.0.4
exceptiongroup                1.1.1
executing                     1.2.0
fairscale                     0.4.0
fastapi                       0.110.0
fastjsonschema                2.17.1
ffmpy                         0.3.1
filelock                      3.12.0
flit_core                     3.9.0
fonttools                     4.49.0
fsspec                        2023.5.0
google-auth                   2.28.1
google-auth-oauthlib          1.2.0
gradio                        4.19.2
gradio_client                 0.10.1
grpcio                        1.62.0
h11                           0.14.0
hjson                         3.1.0
httpcore                      1.0.2
httpx                         0.26.0
huggingface-hub               0.20.3
idna                          3.4
importlib-metadata            6.6.0
importlib-resources           5.12.0
ipykernel                     6.15.0
ipython                       8.14.0
ipython-genutils              0.2.0
ipywidgets                    8.0.6
jedi                          0.18.2
jieba                         0.42.1
Jinja2                        3.1.2
joblib                        1.3.2
json5                         0.9.5
jsonlines                     4.0.0
jsonpatch                     1.32
jsonpointer                   2.1
jsonschema                    4.17.3
jupyter                       1.0.0
jupyter_client                8.2.0
jupyter-console               6.6.3
jupyter_core                  5.3.0
jupyter-events                0.6.3
jupyter-lsp                   2.2.0
jupyter_server                2.6.0
jupyter_server_terminals      0.4.4
jupyterlab                    4.0.1
jupyterlab-pygments           0.2.2
jupyterlab_server             2.22.1
jupyterlab-widgets            3.0.7
kiwisolver                    1.4.5
LAC                           2.1.2
langcodes                     3.3.0
lit                           16.0.5
lxml                          5.1.0
Markdown                      3.5.2
markdown-it-py                3.0.0
markdown2                     2.4.10
MarkupSafe                    2.1.2
matplotlib                    3.7.4
matplotlib-inline             0.1.6
mdurl                         0.1.2
mistune                       2.0.5
more-itertools                10.1.0
mpi4py                        3.1.4
mpmath                        1.3.0
murmurhash                    1.0.10
nbclassic                     1.0.0
nbclient                      0.8.0
nbconvert                     7.4.0
nbformat                      5.9.0
nest-asyncio                  1.5.6
networkx                      3.1
ninja                         1.11.1
nltk                          3.8.1
notebook                      6.5.4
notebook_shim                 0.2.3
numpy                         1.24.4
nvidia-cublas-cu11            11.10.3.66
nvidia-cublas-cu12            12.1.3.1
nvidia-cuda-cupti-cu11        11.7.101
nvidia-cuda-cupti-cu12        12.1.105
nvidia-cuda-nvrtc-cu11        11.7.99
nvidia-cuda-nvrtc-cu12        12.1.105
nvidia-cuda-runtime-cu11      11.7.99
nvidia-cuda-runtime-cu12      12.1.105
nvidia-cudnn-cu11             8.5.0.96
nvidia-cudnn-cu12             8.9.2.26
nvidia-cufft-cu11             10.9.0.58
nvidia-cufft-cu12             11.0.2.54
nvidia-curand-cu11            10.2.10.91
nvidia-curand-cu12            10.3.2.106
nvidia-cusolver-cu11          11.4.0.1
nvidia-cusolver-cu12          11.4.5.107
nvidia-cusparse-cu11          11.7.4.91
nvidia-cusparse-cu12          12.1.0.106
nvidia-ml-py                  12.535.133
nvidia-nccl-cu11              2.14.3
nvidia-nccl-cu12              2.18.1
nvidia-nvjitlink-cu12         12.3.101
nvidia-nvtx-cu11              11.7.91
nvidia-nvtx-cu12              12.1.105
oauthlib                      3.2.2
opencv-python-headless        4.5.5.64
openpyxl                      3.1.2
opt-einsum                    3.3.0
orjson                        3.9.15
overrides                     7.3.1
packaging                     23.2
paddle-bfloat                 0.1.7
paddlepaddle                  2.4.2
pandas                        2.2.1
pandocfilters                 1.5.0
parso                         0.8.3
peft                          0.9.0
pexpect                       4.8.0
pickleshare                   0.7.5
Pillow                        10.1.0
pip                           24.0
pkgutil_resolve_name          1.3.10
platformdirs                  3.5.1
pluggy                        1.0.0
ply                           3.11
portalocker                   2.8.2
preshed                       3.0.9
prometheus-client             0.17.0
prompt-toolkit                3.0.38
protobuf                      4.25.0
psutil                        5.9.5
ptyprocess                    0.7.0
pure-eval                     0.2.2
py-cpuinfo                    9.0.0
pyasn1                        0.5.1
pyasn1-modules                0.3.0
pybind11                      2.10.4
pycosat                       0.6.4
pycparser                     2.21
pydantic                      1.10.14
pydantic_core                 2.16.3
pydub                         0.25.1
Pygments                      2.15.1
pynvml                        11.4.1
pyOpenSSL                     23.0.0
pyparsing                     3.1.1
PyQt5-sip                     12.11.0
pyrsistent                    0.18.0
PySocks                       1.7.1
python-dateutil               2.8.2
python-etcd                   0.4.5
python-json-logger            2.0.7
python-multipart              0.0.9
pytz                          2023.3
PyYAML                        6.0
pyzmq                         25.0.2
qtconsole                     5.4.3
QtPy                          2.3.1
regex                         2023.5.5
requests                      2.28.1
requests-oauthlib             1.3.1
rfc3339-validator             0.1.4
rfc3986-validator             0.1.1
rich                          13.7.0
rsa                           4.9
ruamel.yaml                   0.17.21
ruamel.yaml.clib              0.2.6
ruff                          0.2.2
sacrebleu                     2.3.2
safetensors                   0.4.2
seaborn                       0.13.0
semantic-version              2.10.0
Send2Trash                    1.8.2
sentencepiece                 0.1.99
setuptools                    65.6.3
shellingham                   1.5.4
shortuuid                     1.0.11
sip                           6.6.2
six                           1.16.0
smart-open                    6.4.0
sniffio                       1.3.0
socksio                       1.0.0
soupsieve                     2.3.2.post1
spacy                         3.7.2
spacy-legacy                  3.0.12
spacy-loggers                 1.0.5
srsly                         2.4.8
stack-data                    0.6.2
starlette                     0.36.3
sympy                         1.12
tabulate                      0.9.0
tensorboard                   2.15.2
tensorboard-data-server       0.7.2
terminado                     0.17.1
thinc                         8.2.3
timm                          0.9.10
tinycss2                      1.2.1
titan-plugins                 0.0.3
tokenizers                    0.19.1
toml                          0.10.2
tomli                         2.0.1
tomlkit                       0.12.0
toolz                         0.12.0
torch                         2.1.2+cu118
torchaudio                    2.1.2+cu118
torchvision                   0.16.2+cu118
tornado                       6.2
tqdm                          4.66.1
traitlets                     5.9.0
transformers                  4.40.0
triton                        2.1.0
typer                         0.9.0
typing_extensions             4.6.3
typing-utils                  0.1.0
tzdata                        2024.1
urllib3                       1.26.15
uvicorn                       0.24.0.post1
wasabi                        1.1.2
wcwidth                       0.2.6
weasel                        0.3.4
webencodings                  0.5.1
websocket-client              1.5.2
websockets                    11.0.3
Werkzeug                      3.0.1
wheel                         0.38.4
widgetsnbextension            4.0.7
zipp                          3.15.0
zstandard                     0.19.0

**Comment by Fg-droid-1014** [comment]
非常感谢！请问下面这些是选择11或12还是都需要？

nvidia-cublas-cu11 11.10.3.66
nvidia-cublas-cu12 12.1.3.1
nvidia-cuda-cupti-cu11 11.7.101
nvidia-cuda-cupti-cu12 12.1.105
nvidia-cuda-nvrtc-cu11 11.7.99
nvidia-cuda-nvrtc-cu12 12.1.105
nvidia-cuda-runtime-cu11 11.7.99
nvidia-cuda-runtime-cu12 12.1.105
nvidia-cudnn-cu11 8.5.0.96
nvidia-cudnn-cu12 8.9.2.26
nvidia-cufft-cu11 10.9.0.58
nvidia-cufft-cu12 11.0.2.54
nvidia-curand-cu11 10.2.10.91
nvidia-curand-cu12 10.3.2.106
nvidia-cusolver-cu11 11.4.0.1
nvidia-cusolver-cu12 11.4.5.107
nvidia-cusparse-cu11 11.7.4.91
nvidia-cusparse-cu12 12.1.0.106

**Comment by 1SingleFeng** [comment]
> 非常感谢！请问下面这些是选择11或12还是都需要？
> 
> nvidia-cublas-cu11 11.10.3.66 nvidia-cublas-cu12 12.1.3.1 nvidia-cuda-cupti-cu11 11.7.101 nvidia-cuda-cupti-cu12 12.1.105 nvidia-cuda-nvrtc-cu11 11.7.99 nvidia-cuda-nvrtc-cu12 12.1.105 nvidia-cuda-runtime-cu11 11.7.99 nvidia-cuda-runtime-cu12 12.1.105 nvidia-cudnn-cu11 8.5.0.96 nvidia-cudnn-cu12 8.9.2.26 nvidia-cufft-cu11 10.9.0.58 nvidia-cufft-cu12 11.0.2.54 nvidia-curand-cu11 10.2.10.91 nvidia-curand-cu12 10.3.2.106 nvidia-cusolver-cu11 11.4.0.1 nvidia-cusolver-cu12 11.4.5.107 nvidia-cusparse-cu11 11.7.4.91 nvidia-cusparse-cu12 12.1.0.106

我的cuda版本是11.8，正常来说cuda 11和cuda 12应该都是支持的

**Comment by 1SingleFeng** [comment]
@Fg-droid-1014 如果你要训练，peft需要升级到最新版本

**Comment by Falkonar** [comment]
what pytorch should I use for cuda 12 I want to use it with WSL2
